File: pbmx_ffi/java/io/rmf/pbmx/ffi/RawShare.java
package io.rmf.pbmx.ffi;

import java.util.Arrays;
import java.util.List;
import com.sun.jna.Structure;

public class RawShare extends Structure {
    public static class ByValue extends RawShare implements Structure.ByValue {}

    public byte[] bytes = new byte[32];

    public ByValue val() {
        ByValue value = new ByValue();
        value.bytes = this.bytes;
        return value;
    }

    public RawShare ref() {
        RawShare s = new RawShare();
        s.bytes = this.bytes;
        return s;
    }

    protected List getFieldOrder() {
        return Arrays.asList(new String[] { "bytes" });
    }
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/ffi/RawShare.kt
package io.rmf.pbmx.ffi

import com.sun.jna.Structure

open class RawShare : Structure() {

    @JvmField var bytes = ByteArray(32)

    class ByValue : RawShare(), Structure.ByValue

    fun value(): ByValue {
        val value = ByValue()
        value.bytes = this.bytes
        return value
    }

    override fun getFieldOrder(): List<String> {
        return listOf("bytes")
    }
}
